gnucash/register/register-gnome/combocell-gnome.c

```diff
@@ -534,37 +555,99 @@ gnc_combo_cell_modify_verify (BasicCell *_cell,
     }
 
```

Comment:
Nearly everything from here on is wrapped in a `type_ahead_search` conditional. I think it would be clearer to instead have two helper functions, one for type-ahead and one not.

Reply:
agreed. I'll address your other comments in the updated code. Thanks.